src/custom-elements/src/components/search-input/search-input.tsx

```diff
@@ -139,11 +143,16 @@ export class SearchInput {
       onChange={(ev) => this._handleValueChange(ev)}
       autocomplete="off"
       aria-label="search input"
+      readonly={this.readonly}
     />;
 
     const searchXIcon = this._cleanButtonVisible ?
       <button class="chi-button -icon -close -xs"
-        onClick={() => this._cleanInput()}
+        onClick={() => {
+          if (this.readonly) {
+            this._cleanInput
+          }
```

Comment:
looks like `this._cleanInput` doesn't do anything, it should be `this._cleanInput()`

Reply:
Good catch! Thanks!